src/main/java/frc/robot/subsystems/Turret.java

```diff
@@ -0,0 +1,103 @@
+package frc.robot.subsystems;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.can.WPI_TalonSRX;
+
+import edu.wpi.first.wpilibj.AnalogPotentiometer;
+import edu.wpi.first.wpilibj.command.PIDSubsystem;
+import edu.wpi.first.wpilibj.interfaces.Potentiometer;
+import frc.robot.Config;
+import frc.robot.Logger;
+import frc.robot.Robot;
+import frc.robot.Config.Key;
+import frc.robot.OI.OperatorInterface;
+
+/**
+ * This Turret class extends WPILIB's PID subsystem via @Overriding methods
+ * use enable(), disable() and setSetpoint() to control the PID.
+ * loop() should be run every tick
+ */
+public class Turret extends PIDSubsystem {
+  private Logger mTurretLogger;
+  private WPI_TalonSRX mTurretTalon;
+  private Potentiometer mPot;
+
+  // the target position (on a scale from 0 to 100)
+  private double mManualTargetPos = 50;
+
+  /**
+   * Set up our talon, logger and potentiometer
+   */
+  public Turret() {
```

Comment:
This is not a singleton. All subsystems should be singletons.

Reply:
done